dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/loop/template/http/parser/HttpTaskDefinitionParser.java

```diff
@@ -60,9 +60,9 @@ public class HttpTaskDefinitionParser implements TaskDefinitionParser<HttpLoopTa
     }
 
     protected @NonNull LoopTaskYamlDefinition parseYamlConfigFile(@NonNull String yamlConfigFile) throws IOException {
-        Yaml yaml = new Yaml(new Constructor(LoopTaskYamlDefinition.class));
         try (FileReader fileReader = new FileReader(yamlConfigFile)) {
-            return yaml.load(fileReader);
+            return new Yaml(new ClassFilterConstructor(new Class[]{LoopTaskYamlDefinition.class}))
```

Comment:
I think, you need to add all nested types(except for the primitive types in Java), such as 

https://github.com/apache/dolphinscheduler/blob/2e169076dc565f194c11ee4f55a4ac7f65071c6a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/loop/template/LoopTaskYamlDefinition.java#L29

https://github.com/apache/dolphinscheduler/blob/2e169076dc565f194c11ee4f55a4ac7f65071c6a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/loop/template/LoopTaskYamlDefinition.java#L36


https://github.com/apache/dolphinscheduler/blob/2e169076dc565f194c11ee4f55a4ac7f65071c6a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/loop/template/LoopTaskYamlDefinition.java#L42

And so on……

Reply:
@kezhenxu94 In `ClassFilterConstructor`, it overrides the method `getClassForName` from its super class `Constructor` which is called in the method `getClassForNode`.  The strange thing is that if you put a check point at `cl = this.getClassForName(name);`, run `HttpTaskDefinitionParserTest.parseYamlConfigFile` and you will find that `cl = this.getClassForName(name);` only get called once, which means the fields and the fields of the fields in `LoopTaskYamlDefinition` such as `LoopTaskServiceYamlDefinition`, `LoopTaskQueryStateYamlDefinition`, etc. are not checked iteratively. If I add the nested types, I think whether to add these nested types or not does not make any differences and the nested types still bypass the check in this solution.

``` java
    protected Class<?> getClassForNode(Node node) {
        Class<? extends Object> classForTag = (Class)this.typeTags.get(node.getTag());
        if (classForTag == null) {
            String name = node.getTag().getClassName();

            Class cl;
            try {
                cl = this.getClassForName(name);
            } catch (ClassNotFoundException var6) {
                throw new YAMLException("Class not found: " + name);
            }

            this.typeTags.put(node.getTag(), cl);
            return cl;
        } else {
            return classForTag;
        }
    }
```


![image](https://github.com/apache/dolphinscheduler/assets/34905992/e684ff4e-57a9-423c-bb8b-cbeb84b2952d)

Reply:
> @kezhenxu94 In `ClassFilterConstructor`, it overrides the method `getClassForName` from its super class `Constructor` which is called in the method `getClassForNode`. The strange thing is that if you put a check point at `cl = this.getClassForName(name);`, run `HttpTaskDefinitionParserTest.parseYamlConfigFile` and you will find that `cl = this.getClassForName(name);` only gets called once, which means the fields and the fields of the fields in `LoopTaskYamlDefinition` such as `LoopTaskServiceYamlDefinition`, `LoopTaskQueryStateYamlDefinition`, etc. are not checked iteratively. I think whether to add these nested types or not does not make any difference and the nested types still bypass the check in this solution.
> 
> ```java
>     protected Class<?> getClassForNode(Node node) {
>         Class<? extends Object> classForTag = (Class)this.typeTags.get(node.getTag());
>         if (classForTag == null) {
>             String name = node.getTag().getClassName();
> 
>             Class cl;
>             try {
>                 cl = this.getClassForName(name);
>             } catch (ClassNotFoundException var6) {
>                 throw new YAMLException("Class not found: " + name);
>             }
> 
>             this.typeTags.put(node.getTag(), cl);
>             return cl;
>         } else {
>             return classForTag;
>         }
>     }
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/34905992/316373126-e684ff4e-57a9-423c-bb8b-cbeb84b2952d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzMzY0MjksIm5iZiI6MTcxMTMzNjEyOSwicGF0aCI6Ii8zNDkwNTk5Mi8zMTYzNzMxMjYtZTY4NGZmNGUtNTdhOS00MjNjLWJiOGItY2JlYjg0YjI5NTJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDAzMDg0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViOTI5ODdmMjJhOGFhODJlYmNkZTcxMWQ4ZDIxNDk1ZmJhOTMzOWE5MzdlOTUzMWUwMTY0OTdiMTczMmZlMGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lctEfSV1UeCvDHl2fSvGcXCc1VtGRdeu9itW50u5A7U)

Reply:
I think the method `getClassForNode` must be called per yaml section node, it is impossible (to me) to get all nested classes types when starting to parse the root node, will check.